File: modules/react-native-sample/android/src/main/java/com/sample/entity/Quote.kt
package com.sample.entity

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.WritableMap

data class Quote(
    override val key: String,
    val symbolName: String,
    val companyName: String,
    val tradePrice: String,
    val tradePriceStatus: QuoteBaseStatus,
    val point: String,
    val pointStatus: QuoteBaseStatus,
    val percent: String,
    val percentStatus: QuoteBaseStatus,
    override val isFooter: Boolean,
    override val isHeader: Boolean
) : BaseEntity {

    fun toMap(): WritableMap {
        return Arguments.createMap().apply {
            putString("key", key)
            putString("symbolName", symbolName)
            putString("companyName", companyName)
            putString("tradePrice", tradePrice)
            putString("tradePriceStatus", tradePriceStatus.toString())
            putString("point", point)
            putString("pointStatus", pointStatus.toString())
            putString("percent", percent)
            putString("percentStatus", percentStatus.toString())
        }
    }

    companion object {
        fun createHeaderFooter(isHeader: Boolean): Quote {
            return Quote(
                key = if (isHeader) {
                    "HEADER"
                } else {
                    "FOOTER"
                },
                symbolName = "",
                companyName = "",
                tradePrice = "",
                tradePriceStatus = QuoteBaseStatus.UNKNOWN,
                point = "",
                pointStatus = QuoteBaseStatus.UNKNOWN,
                percent = "",
                percentStatus = QuoteBaseStatus.UNKNOWN,
                isFooter = !isHeader,
                isHeader = isHeader
            )
        }

        fun fromArray(array: ReadableArray): ArrayList<Quote> {
            val resultArray = arrayListOf<Quote>()
            for (i in 0 until array.size()) {
                val map = array.getMap(i)
                resultArray.add(fromMap(map))
            }
            return resultArray
        }

        fun fromMap(map: ReadableMap): Quote {
            val key = map.getString("key") ?: ""
            val symbolName = map.getString("symbolName") ?: ""
            val companyName = map.getString("companyName") ?: ""
            val tradePrice = map.getString("tradePrice") ?: ""
            val tradePriceStatus =
                QuoteBaseStatus.fromString(map.getString("tradePriceStatus") ?: "")
            val point = map.getString("point") ?: ""
            val pointStatus = QuoteBaseStatus.fromString(map.getString("pointStatus") ?: "")
            val percent = map.getString("percent") ?: ""
            val percentStatus = QuoteBaseStatus.fromString(map.getString("percentStatus") ?: "")
            val isFooter = false
            val isHeader = false

            return Quote(
                key = key,
                symbolName = symbolName,
                companyName = companyName,
                tradePrice = tradePrice,
                tradePriceStatus = tradePriceStatus,
                point = point,
                pointStatus = pointStatus,
                percent = percent,
                percentStatus = percentStatus,
                isFooter = isFooter,
                isHeader = isHeader
            )
        }
    }
}

File: modules/react-native-sample/ios/Entity/Quote.swift
struct Quote {
    let key                 : String
    let symbolName          : String
    let companyName         : String
    let tradePrice          : String
    let tradePriceStatus    : QuoteBaseStatus
    let pointStatus         : QuoteBaseStatus
    let percentStatus       : QuoteBaseStatus
    let point               : String
    let percent         :    String
    
    
    static func dic2Quote(dic: NSDictionary) -> Quote {
        let key                 = dic["key"]                as? String  ?? ""
        let symbolName          = dic["symbolName"]         as? String  ?? ""
        let companyName         = dic["companyName"]        as? String  ?? ""
        let tradePrice          = dic["tradePrice"]         as? String  ?? ""
        let tradePriceStatus    = dic["tradePriceStatus"]   as? String  ?? ""
        let pointStatus         = dic["pointStatus"]        as? String  ?? ""
        let percentStatus       = dic["percentStatus"]      as? String  ?? ""
        let point               = dic["point"]              as? String  ?? ""
        let percent             = dic["percent"]            as? String  ?? ""
        
        return Quote(key: key,
                     symbolName         : symbolName,
                     companyName        : companyName,
                     tradePrice         : tradePrice,
                     tradePriceStatus   : QuoteBaseStatus.fromString(value: tradePriceStatus),
                     pointStatus        : QuoteBaseStatus.fromString(value: pointStatus),
                     percentStatus      : QuoteBaseStatus.fromString(value: percentStatus),
                     point              : point,
                     percent            : percent
        )
    }
    
    static func arrayDicToArrayQuote(arr: Array<NSDictionary>) -> Array<Quote> {
        return arr.map(Quote.dic2Quote)
    }
}

extension Quote {
    func toDic() -> [String: Any] {
        return [
            "key"               : key,
            "symbolName"        : symbolName,
            "companyName"       : companyName,
            "tradePrice"        : tradePrice,
            "tradePriceStatus"  : tradePriceStatus.rawValue,
            "pointStatus"       : pointStatus.rawValue,
            "percentStatus"     : percentStatus.rawValue,
            "point"             : point,
            "percent"           : percent
        ]
    }
}
